portal/server/app/route.ts

```diff
@@ -4,16 +4,32 @@
 import { getDomain, getSubdomainAndPath } from "@lib/domain_parsing";
 import { redirectToAggregatorUrlResponse, redirectToPortalURLResponse } from "@lib/redirects";
 import { getBlobIdLink, getObjectIdLink } from "@lib/links";
-import { resolveAndFetchPage } from "@lib/page_fetching";
+import { PageFetcher } from "@lib/page_fetching";
+import { ResourceFetcher } from "@lib/resource";
+import { RPCSelector } from "@lib/rpc_selector";
+
 import { siteNotFound } from "@lib/http/http_error_responses";
 import integrateLoggerWithSentry from "sentry_logger";
 import blocklistChecker from "custom_blocklist_checker";
+import { SuiNSResolver } from "@lib/suins";
+import { WalrusSitesRouter } from "@lib/routing";
 
 if (process.env.ENABLE_SENTRY === "true") {
     // Only integrate Sentry on production.
     integrateLoggerWithSentry();
 }
 
+const rpcUrlList = process.env.RPC_URL_LIST;
+if (!rpcUrlList) {
+    throw new Error("Missing RPC_URL_LIST environment variable");
+}
+const rpcSelector = new RPCSelector(rpcUrlList.split(','));
+const pageFetcher = new PageFetcher(
+    new ResourceFetcher(rpcSelector),
+    new SuiNSResolver(rpcSelector),
+    new WalrusSitesRouter(rpcSelector)
+);
+
```

Comment:
Praise: yesss I like this
Question: in TS/JS, is this passed by reference or by value? I.e., can we be sure that the selector is the same for all three objects afterwards?

Reply:
Question: do we still need this as a global singleton now?

Reply:
It's no longer a singleton in its' strict sense, meaning that its' constructor is no longer private, and multiple `rpcSelectors` can be created inside the portal. 

More specifically on these I am instantiating the `rpcSelector` once and pass it by reference to the other classes' constructors.

Reply:
ok, very interesting thanks!

I'm not familiar with these global `const`s, and i'm not sure if/how they could be problematic. but looks good for now i guess.